src/utils/validatePassword.js

```diff
@@ -34,7 +34,7 @@ function validatePassword(password) {
   if (hasNumber && !/[0-9]/.test(password)) {
     errors.push("Password must contain at least one number.");
   }
-  if (hasSpecialChar && !/[!@#$%^&*\-_]/.test(password)) {
+  if (hasSpecialChar && !/[!?~@#$%'^"&*[]\-_]/.test(password)) {
```

Comment:
This regular expression doesn't work as intended -- you can test it at [regex101](https://regex101.com/?regex=%5B!%3F~%40%23%24%25'%5E%22%26*%5B%5D%5C-_%5D&flags=&testString=ThisStringHasSpecialCharacters%5D%5C-&flavor=javascript).

Some characters need to be escaped with a `\`.

Reply:
fixed